internal/engine/buildcontrol/build_control.go

```diff
@@ -19,8 +19,9 @@ func NextTargetToBuild(state store.EngineState) *store.ManifestTarget {
 		return nil
 	}
 
-	targets := RemoveCurrentlyBuildingTargets(state.Targets())
-	targets = RemoveTargetsWaitingOnDependencies(state, targets)
+	sortedTargets := TopologicalSort(state.Targets())
+	sortedTargets = RemoveCurrentlyBuildingTargetsAndDeps(sortedTargets)
+	targets := RemoveTargetsWaitingOnDependencies(state, []*store.ManifestTarget(sortedTargets))
```

Comment:
imo `RemoveCurrentlyBuildingTargetsAndDeps` and `RemoveTargetsWaitingOnDependencies` aren't very distinct, the difference between them isn't apparent at a glance (tho this might just be me)